rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/v3/VersionedModelControllerV3.scala

```diff
@@ -0,0 +1,217 @@
+import za.co.absa.enceladus.model.versionedModel._
+import za.co.absa.enceladus.model.{ExportableObject, UsedIn, Validation}
+import za.co.absa.enceladus.rest_api.controllers.BaseController
+import za.co.absa.enceladus.rest_api.controllers.v3.VersionedModelControllerV3.LatestVersionKey
+import za.co.absa.enceladus.rest_api.services.VersionedModelService
+
+import java.net.URI
+import java.util
+import java.util.Optional
+import java.util.concurrent.CompletableFuture
+import javax.servlet.http.HttpServletRequest
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+abstract class VersionedModelControllerV3[C <: VersionedModel with Product
+  with Auditable[C]](versionedModelService: VersionedModelService[C]) extends BaseController {
+
+  import za.co.absa.enceladus.rest_api.utils.implicits._
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  // todo maybe offset/limit?
+  @GetMapping(Array(""))
+  @ResponseStatus(HttpStatus.OK)
+  def getList(@RequestParam searchQuery: Optional[String]): CompletableFuture[Seq[VersionedSummary]] = {
+    versionedModelService.getLatestVersionsSummary(searchQuery.toScalaOption)
+  }
+
+  @GetMapping(Array("/{name}"))
+  @ResponseStatus(HttpStatus.OK)
+  def getVersionsList(@PathVariable name: String): CompletableFuture[VersionsList] = {
+    versionedModelService.getAllVersionsValues(name) map {
+      case Some(entity) => entity
+      case None => throw notFound()
+    }
+  }
+
+  @GetMapping(Array("/{name}/{version}"))
+  @ResponseStatus(HttpStatus.OK)
+  def getVersionDetail(@PathVariable name: String,
+                       @PathVariable version: String): CompletableFuture[C] = {
+    forVersionExpression(name, version)(versionedModelService.getVersion).map {
+      case Some(entity) => entity
+      case None => throw notFound()
+    }
+  }
+
+
+  @GetMapping(Array("/{name}/audit-trail"))
+  @ResponseStatus(HttpStatus.OK)
+  def getAuditTrail(@PathVariable name: String): CompletableFuture[AuditTrail] = {
+    versionedModelService.getAuditTrail(name)
+  }
+
+  @GetMapping(Array("/{name}/{version}/used-in"))
+  @ResponseStatus(HttpStatus.OK)
+  def usedIn(@PathVariable name: String,
+             @PathVariable version: String): CompletableFuture[UsedIn] = {
+    forVersionExpression(name, version) { case (name, versionInt) => versionedModelService.getUsedIn(name, Some(versionInt)) }
+  }
+
+  @GetMapping(Array("/{name}/{version}/export"))
+  @ResponseStatus(HttpStatus.OK)
+  def exportSingleEntity(@PathVariable name: String, @PathVariable version: String): CompletableFuture[String] = {
+    forVersionExpression(name, version)(versionedModelService.exportSingleItem)
+  }
+
+  @GetMapping(Array("/{name}/export"))
+  @ResponseStatus(HttpStatus.OK)
+  def exportLatestEntity(@PathVariable name: String): CompletableFuture[String] = {
+    versionedModelService.exportLatestItem(name) // todo: remove in favor of the above? (that supports /{name}/latest/export)
```

Comment:
From my point of view, this can stay

Reply:
Well, this is part of the v3 API outline. I consider this misleading, because since you have the option to export specific versions already ( at .../{model}/{name}/{versionEitherNumberOrLatestKeyword}/export, having the option for export here would have to have another meaning - e.g. export the whole entity with all the versions. But since this is designed as latest-export as well, I consider this unnecessary, and as I said, even misleading.

That is why I am proposing to omit this endpoint entirely.

Reply:
After even further discussion, I have decided to remove the endpoint based on the following logic:
The structure of API endpoints representing a collection should, in general, behave in the following fashion:
 - `GET /entityNamePlural` -> list items or provide information about all/more items
 - `GET /entityNamePlural/individualEntityKeyEgId` -> give information about a specific entity

By that logic, `GET /{name}/export` would be appropriate if all entities or information about all/more was returned -- perhaps all versions, while `GET /{name}/{version}/export` would export just a single version.

But since we mapped just the single version (the latest one) export to a collection's endpoint, the logic does not fit. Moreover, given the fact that we also accept `latest` as a valid version identifier, it seem much more appropriate to use `GET /{name}/{version}/export` here (specifically `GET /{name}/latest/export`)